eZ/Publish/Core/Persistence/Cache/LocationHandler.php

```diff
@@ -327,4 +334,37 @@ public function loadAllLocations($offset, $limit)
 
         return $this->persistenceHandler->locationHandler()->loadAllLocations($offset, $limit);
     }
+
+    /**
+     * Return relevant content and location tags so cache can be purged reliably.
+     *
+     * @param Location $location
+     * @param array $tags Optional, can be used to specify additional tags.
+     *
+     * @return array
+     */
+    private function getCacheTags(Location $location, $tags = [])
+    {
+        $tags[] = 'content-' . $location->contentId;
+        $tags[] = 'location-' . $location->id;
+        $tags[] = 'location-data-' . $location->id;
+        foreach (explode('/', trim($location->pathString, '/')) as $pathId) {
+            $tags[] = 'location-path-' . $pathId;
+            $tags[] = 'location-path-data-' . $pathId;
+        }
+
+        return $tags;
+    }
+
+    private function getCacheTranslationKey(array $translations = null, bool $useAlwaysAvailable = true): string
+    {
+        if (empty($translations)) {
+            return (int)$useAlwaysAvailable;
+        }
+
+        // Sort array as we don't care about order in location handler usage & want to optimize for cache hits.
+        sort($translations);
+
+        return implode('|', $translations) . '|' . (int)$useAlwaysAvailable;
```

Comment:
@alongosz this is an interesting questions. How many of the elements that make up cache-keys used by eZ actually do allow a dash character as valid ?
Eg.
- using content-id as cache key => safe
- using content type identifier => most likely safe in ez4, which only allowed underscores, but not in ezplatform

The risk is of cours cache key collision:

my-key - another === my - another-key

Reply:
@alongosz I could change it, but be aware this is more or less copy paste from contentHandler, only difference is it does not handle always available like we do here*

* There is a _different_ discussion to be had on that too, I'm starting to think we should pass that down to SPI in contentHandler too, but it would then imply we filter on language mask there to, and not just languageCode in attribute table.

Reply:
>  I could change it, but be aware this is more or less copy paste from contentHandler, only difference is it does not handle always available like we do here

No problem, leaving then decision up to you :)